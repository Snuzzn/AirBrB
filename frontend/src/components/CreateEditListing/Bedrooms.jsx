import React from 'react'
import PropTypes from 'prop-types';
import { BsPlusCircle } from 'react-icons/bs';

function Bedrooms ({ bedrooms, setBedrooms }) {
  const handleBedroomChange = (e, index, isTitle) => {
    const input = e.target.value
    const old = [...bedrooms]
    if (isTitle) old[index].title = input
    else old[index].count = input
    setBedrooms(old)
  }

  return (
    <div className="flex flex-col gap-2 justify-center">
      <div className="flex items-center gap-3 mb-2">
        <label>Bedrooms</label>
        <BsPlusCircle onClick={() => setBedrooms([...bedrooms, { title: '', count: 0 }])} className=" text-red-400 cursor-pointer hover:text-red-500" />
      </div>
      {bedrooms.map((item, index) => (
        <div key={index} className="flex gap-5 items-center ">
          <div className="flex flex-col gap-1 justify-center">
            <label className="text-sm text-gray-600">Name</label>
            <input type="text" value={bedrooms[index].title} onChange={(e) => handleBedroomChange(e, index, true)} required className="p-2 border border-gray-300 rounded-lg" />
          </div>
          <div className="flex flex-col gap-1 justify-center">
            <label className="text-sm text-gray-600">No. of beds</label>
            <input type="number" value={bedrooms[index].count} onChange={(e) => handleBedroomChange(e, index, false)} required className="p-2 border border-gray-300 rounded-lg w-24" />
          </div>
        </div>
      ))}
    </div>
  )
}

export default Bedrooms

Bedrooms.propTypes = {
  bedrooms: PropTypes.string,
  setBedrooms: PropTypes.string
}
