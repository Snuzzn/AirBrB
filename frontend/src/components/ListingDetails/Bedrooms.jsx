import React from 'react'
import { IoBedOutline } from 'react-icons/io5'
import PropTypes from 'prop-types';

function Bedrooms ({ listingInfo }) {
  let bedrooms = []
  if (listingInfo.metadata != null) bedrooms = listingInfo.metadata.bedrooms
  return (
    <div className="flex gap-4">
      { bedrooms.map((bedroom) =>
        <React.Fragment key={bedroom.title}>
          <div className="h-40 w-60 border p-4 rounded-xl flex flex-col justify-evenly">
            <IoBedOutline className="text-5xl text-gray-600"/>
            <div>
              <p className="text-lg text-gray-700" id={bedroom.title}>{bedroom.title}</p>
              <p className="text-gray-600 font-light" id={bedroom.title + 'count'}>{bedroom.count} beds</p>
            </div>
          </div>
        </React.Fragment>
      )}
    </div>
  )
}

export default Bedrooms

Bedrooms.propTypes = {
  listingInfo: PropTypes.object,
}
