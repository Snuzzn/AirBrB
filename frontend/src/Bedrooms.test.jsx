import React from 'react';
import { shallow } from 'enzyme';
import { IoBedOutline } from 'react-icons/io5'
import Bedrooms from './components/ListingDetails/Bedrooms';

describe('Bedrooms', () => {
  /**
   * BookingInfoCard takes title, message, icon, and background colour to
   * display the appropriate metric. It is used in the Booking Details page.
   */
  const listingInfo = { metadata: { bedrooms: [{ title: 'Room1', count: 2 }, { title: 'Room2', count: 4 }] } }

  it('renders 1st bedroom title', () => {
    const bedrooms = shallow(<Bedrooms listingInfo={listingInfo} />)
    const bedroom = bedrooms.find('#Room1').text()
    expect(bedroom).toEqual('Room1')
  })

  it('renders 1st bedroom count', () => {
    const bedrooms = shallow(<Bedrooms listingInfo={listingInfo} />)
    const bedroom = bedrooms.find('#Room1count').text()
    expect(bedroom).toEqual('2 beds')
  })

  it('renders 2nd bedroom title', () => {
    const bedrooms = shallow(<Bedrooms listingInfo={listingInfo} />)
    const bedroom = bedrooms.find('#Room2').text()
    expect(bedroom).toEqual('Room2')
  })

  it('renders 2nd bedroom count', () => {
    const bedrooms = shallow(<Bedrooms listingInfo={listingInfo} />)
    const bedroom = bedrooms.find('#Room2count').text()
    expect(bedroom).toEqual('4 beds')
  })

  it('contains a bed icon', () => {
    const bedrooms = shallow(<Bedrooms listingInfo={listingInfo} />)
    const icon = <IoBedOutline className="text-5xl text-gray-600"/>
    expect(bedrooms.contains(icon)).toEqual(true)
  })

})
