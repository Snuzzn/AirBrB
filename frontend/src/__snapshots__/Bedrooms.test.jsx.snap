// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bedrooms renders all data 1`] = `
<div
  className="flex gap-4"
>
  <div
    className="h-40 w-60 border p-4 rounded-xl flex flex-col justify-evenly"
  >
    <svg
      alt="bed icon"
      className="text-5xl text-gray-600"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M384 240H96V136a40.12 40.12 0 0140-40h240a40.12 40.12 0 0140 40v104zM48 416V304a64.19 64.19 0 0164-64h288a64.19 64.19 0 0164 64v112"
        fill="none"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="32"
      />
      <path
        d="M48 416v-8a24.07 24.07 0 0124-24h368a24.07 24.07 0 0124 24v8M112 240v-16a32.09 32.09 0 0132-32h80a32.09 32.09 0 0132 32v16m0 0v-16a32.09 32.09 0 0132-32h80a32.09 32.09 0 0132 32v16"
        fill="none"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="32"
      />
    </svg>
    <div>
      <p
        className="text-lg text-gray-700"
        id="Room1"
      >
        Room1
      </p>
      <p
        className="text-gray-600 font-light"
        id="Room1count"
      >
        2
         beds
      </p>
    </div>
  </div>
  <div
    className="h-40 w-60 border p-4 rounded-xl flex flex-col justify-evenly"
  >
    <svg
      alt="bed icon"
      className="text-5xl text-gray-600"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M384 240H96V136a40.12 40.12 0 0140-40h240a40.12 40.12 0 0140 40v104zM48 416V304a64.19 64.19 0 0164-64h288a64.19 64.19 0 0164 64v112"
        fill="none"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="32"
      />
      <path
        d="M48 416v-8a24.07 24.07 0 0124-24h368a24.07 24.07 0 0124 24v8M112 240v-16a32.09 32.09 0 0132-32h80a32.09 32.09 0 0132 32v16m0 0v-16a32.09 32.09 0 0132-32h80a32.09 32.09 0 0132 32v16"
        fill="none"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="32"
      />
    </svg>
    <div>
      <p
        className="text-lg text-gray-700"
        id="Room2"
      >
        Room2
      </p>
      <p
        className="text-gray-600 font-light"
        id="Room2count"
      >
        4
         beds
      </p>
    </div>
  </div>
</div>
`;

exports[`Bedrooms renders no props 1`] = `
<div
  className="flex gap-4"
/>
`;
